File: 01jupyter/nb/source/03_for_collapse/problems/00_parallel_for/parallel_for.f90
program parallel_for
  use omp_lib
  integer, parameter :: n = 8
  integer :: a(0:n-1)
  integer :: i
  ! TODO: 下の do ループを !$omp parallel do ... !$omp end parallel do で囲み, 繰り返しを複数のスレッドに分担させよ.
  ! BEGIN ANSWER
  !$omp parallel do
  ! END ANSWER
  do i = 0, n - 1
     a(i) = i * i
     print "(a,i0,a,i0,a,i0,a)", "a[", i, "] = ", a(i), &
          char(9)//"(thread ", omp_get_thread_num(), ")"
  end do
  ! BEGIN ANSWER
  !$omp end parallel do
  ! END ANSWER
end program parallel_for

File: 01jupyter/nb/source/03_for_collapse/problems/00_parallel_for/parallel_for.cpp
#include <cstdio>
#include <omp.h>

int main() {
  const int n = 8;
  int a[n];
  // BEGIN ANSWER: 下の for ループの直前に #pragma omp parallel for を1行追加し, 繰り返しを複数のスレッドに分担させよ.
#pragma omp parallel for
  // END ANSWER
  for (int i = 0; i < n; i++) {
    a[i] = i * i;
    printf("a[%d] = %d\t(thread %d)\n", i, a[i], omp_get_thread_num());
  }
  return 0;
}
